aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp

```diff
@@ -1046,7 +1046,7 @@ Tensor reduce_sparse_csr_dim0_cpu_template(const Tensor& sparse, ReductionOp rop
 
   AT_DISPATCH_INDEX_TYPES(col_indices.scalar_type(), "reduce_sparse_csr_dim0_cpu_indices",
                           [&]() {
-                            index_t* columns_map_ptr = columns_map.data_ptr<index_t>();
+                            int64_t* columns_map_ptr = columns_map.data_ptr<int64_t>();
```

Comment:
Good catch. Could you eliminate using `AT_DISPATCH_INDEX_TYPES` as `index_t` is not used in this block anymore?

Reply:
Thanks for your comments. I find that simply changing `index_t` to `int64_t` here is not that appropriate. The root cause should be `index_type` should be align with `columns_map`'s dtype, not `col_indices`'s dtype. Have updated.